### PR TITLE
Add decode test for llama3.1 model

### PR DESCRIPTION
**Pull request by rsuderman**
Include a test for decode that performs cross entropy across the decode steps.

### CODE REVIEW

**Review by ScottTodd — APPROVED** [review]
Please fix the DCO check: https://github.com/iree-org/iree-test-suites/pull/51/checks?check_run_id=33813712253

Docs: https://iree.dev/developers/general/contributing/#developer-certificate-of-origin

File: sharktank_models/llama3.1/test_llama.py [hunk 1]
```diff
@@ -71,8 +68,49 @@ def __init__(self, compiled, device_id, irpa):
...
+        PADN = N + pad
+
+        ids = numpy.asarray([ids], dtype=numpy.int64)
+        length = numpy.asarray([N], numpy.int64)
+        pages = PADN // block_size
+        pages = numpy.asarray([list(range(pages))], numpy.int64)
+
+        logits = self._prefill(ids[:, :PADN], length, pages, *self._cache)
+        logits = numpy.asarray(logits).astype(numpy.float32)
+        return logits[:, :N]
```
Comment (ScottTodd) [inline sharktank_models/llama3.1/test_llama.py:89]: It isn't obvious to me at a glance what this line is doing. Would a longer variable name for `N` help? Or a comment here on what the slice is accomplishing?

File: sharktank_models/llama3.1/test_llama.py [hunk 2]
```diff
@@ -71,8 +68,49 @@ def __init__(self, compiled, device_id, irpa):
...
+        self._cache = self.allocate_cache()
+
+        self._prefill = self.modules[-1].prefill_bs1
+        self._decode = self.modules[-1].decode_bs1
+
+    def prefill(self, *, ids):
+        N = len(ids)
+        pad = (16 - N % 16) % 16
+        ids = ids + [0] * pad
+        PADN = N + pad
```
Comment (ScottTodd) [inline sharktank_models/llama3.1/test_llama.py:80]: Might be worth a comment here (also, why 16? would 32 or another work?)

```suggestion
        # Pad to the next multiple of 16.
        pad = (16 - N % 16) % 16
        ids = ids + [0] * pad
        PADN = N + pad
```

File: sharktank_models/llama3.1/test_llama.py [hunk 3]
```diff
@@ -106,34 +144,48 @@ def cross_entropy(predictions, targets, epsilon=1e-12):
...
-    logits = numpy.asarray(logits).astype(numpy.float32)
-    ids = numpy.asarray(ids)
+def decode_cross_entropy(toy_llama, ids):
+    N = len(ids)
+    logits = toy_llama.decode(ids=ids, start=0)
+    ids = numpy.asarray([ids])
+    logits = logits[0, :-1, :]
+    ids = ids[:, 1:]
+
+    return cross_entropy(logits, ids)
```
Comment (ScottTodd) [inline sharktank_models/llama3.1/test_llama.py:163]: Out of curiosity, what does `logits` look like before this slicing? What information is discarded?

**Review by ScottTodd — APPROVED** [review]

File: sharktank_models/llama3.1/test_llama.py (hunk 1 above)
Comment (rsuderman) [reply sharktank_models/llama3.1/test_llama.py]: Updated the name to make it make more sense. Its just so we pad out to the blocked sequence length.

File: sharktank_models/llama3.1/test_llama.py (hunk 2 above)
Comment (rsuderman) [reply sharktank_models/llama3.1/test_llama.py]: Added a comment about padding

File: sharktank_models/llama3.1/test_llama.py (hunk 3 above)
Comment (rsuderman) [reply sharktank_models/llama3.1/test_llama.py]: Added a comment to clarify. Its just comparing the predicted to the selected token sequence.
